gdx/src/com/badlogic/gdx/graphics/debugging/GLDebugger.java

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ * Copyright 2015 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.graphics.debugging;
+
+import static com.badlogic.gdx.graphics.GL20.*;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.utils.GdxRuntimeException;
+
+/** Enabling this will wrap Gdx.gl20 and Gdx.gl30 instances into GLDebugger subclasses that check for GL errors after each GL call,
+ * using glGetError. When error happens, {@link GLDebugger#listener} is called inside GL call, so the stack trace may be
+ * inspected.
+ *
+ * @see GL30Debugger
+ * @see GL20Debugger
+ * @see com.badlogic.gdx.graphics.profiling.GLProfiler
+ * @author Jan Polák */
+public abstract class GLDebugger {
+
+	/** Listener that will log using Gdx.app.error GL error number and GL function. */
+	public static final GLDebuggerErrorListener LOGGING_LISTENER = new GLDebuggerErrorListener() {
+		@Override
+		public void onError (int error) {
+			String place = null;
+			try {
+				final StackTraceElement[] stack = Thread.currentThread().getStackTrace();
+				for (int i = 0; i < stack.length; i++) {
+					if ("check".equals(stack[i].getMethodName())) {
+						if (i + 1 < stack.length) {
+							final StackTraceElement glMethod = stack[i + 1];
+							place = glMethod.getMethodName();
+						}

```

Comment:
I am not sure what is the best way to go about this. I have three ideas:
- Emulate whole GLDebugger for GWT, without that one offending line.
  - Most API friendly, but creates a lot of maintenance overhead
- Move LOGGING_LISTENER to a separate class and exclude that from GWT
  - Low maintenance overhead, but not that pretty, I don't like this much
- Remove the "offending gl-call finding" functionality altogether and use only stack trace from `new Exception`, as fallback does now

I am not sure which to choose. Third one is simplest, but I quite like that feature, so I don't want to remove it. (But I will remove it if people think that it is useless) There is also one more alternative, combination of all of them, and that is moving all listeners to a separate class (perhaps as static fields of GLDebuggerErrorListener) and emulating that, such that in GWT version, only what-is-now-fallback is used.

Any thoughts?